src/cli/cli_uart.cpp

```diff
@@ -276,8 +276,14 @@ void Uart::Send(void)
 #if OPENTHREAD_CONFIG_ENABLE_DEBUG_UART
         /* duplicate the output to the debug uart */
         otPlatDebugUart_write_bytes(reinterpret_cast<uint8_t *>(mTxBuffer + mTxHead), mSendLength);
+#endif
+#if OPENTHREAD_UART_LOCK
+        otPlatUartLockAcquire();
```

Comment:
Are we going to allow other process access `otPlatUart*()` API? I think those APIs are defined for OpenThread core. Is guarding the `Uart::Output()` enough? We cannot simply write bytes in NCP mode.

Instead of adding new platform APIs, I think we can also define two macros(default to empty) just like how LWIP does:

```cpp
LOCK_CLI_UART_OUTPUT();
UNLOCK_CLI_UART_OUTPUT();
```